### PR TITLE
Feat/reenter

### PR DESCRIPTION
**Pull request by purpose233**
[[中文版模板 / Chinese template](https://github.com/VisActor/vgrammar/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Chore
- [ ] Release
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/VisActor/VGrammar/issues/103

### 💡 Background and solution

Support mark reenter.
Mark will be marked as reentered when enter encode is updated which would be consumed in the next dataflow and enter encode would be evaluated.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vgrammar/src/view/mark.ts [hunk 1]
```diff
@@ -83,6 +83,9 @@ export class Mark extends GrammarBase implements IMark {
 
   isUpdated: boolean = true;
 
+  /** whether mark enter encode is updated  */
+  isReentered: boolean = false;
```
Comment (xile611) [inline packages/vgrammar/src/view/mark.ts:87]: save this as a private variable?

File: packages/vgrammar/__tests__/mark/element.test.ts [hunk 2]
```diff
@@ -42,6 +42,7 @@ test('Element executes encoding', function () {
     {
       enter: { x: 123 }
     },
+    false,
```
Comment (xile611) [inline packages/vgrammar/__tests__/mark/element.test.ts:45]: maybe you need to add a test case for `reenter`

File: packages/vgrammar/src/view/mark.ts (hunk 1 above)
Comment (purpose233) [reply packages/vgrammar/src/view/mark.ts]: Fixed.

File: packages/vgrammar/__tests__/mark/element.test.ts (hunk 2 above)
Comment (purpose233) [reply packages/vgrammar/__tests__/mark/element.test.ts:45]: Fix. Added in mark.test.ts file.

**Review by xile611 — APPROVED** [review]

**Pull request merged**